### ISSUE TITLE
Userprofile update not possible with username length >60

### ISSUE THREAD

**Issue by lukasassmann** [kimai/kimai#4199]
### Describe the issue

When length of the username in the database is >60 it is not possible to update the profile. An error occurs stating "This value is too long. It should have 60 characters or less.".

In our case, this happens because we use authentik as a SAML provider. The saml provider automatically maps the identifier provided by authentik (which is a 64 character base64 string) to the username. Therefore the user is created with a username > 60 characters in length but can never be edited because of the length restriction.

This can easily be fixed by setting the max-value in Entity/User.php:162 from the current 60 to the actual field length of 180

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.0.28

### How do you run Kimai?

Docker

### Which PHP version are you using?

8.1

### Screenshots

<img width="1914" alt="Screenshot 2023-07-24 at 12 53 08" src="https://github.com/kimai/kimai/assets/139363375/8b2c0e96-a095-42a6-9b20-9712930fe1dd">
<img width="1913" alt="Screenshot 2023-07-24 at 12 52 43" src="https://github.com/kimai/kimai/assets/139363375/a786dec1-581c-460d-980b-5d904f444d5f">

**Comment by kevinpapst** [comment]
Any chance you could send a PR? 
You have the working use-case in front of you and it would be much easier to check that it works if you could do it.
You have to restart php-fm inside docker, to flush the opcache, if you make any chances to PHP code and/attributes

**Comment by lukasassmann** [comment]
Should be fixed with this: https://github.com/kimai/kimai/pull/4200

Tested it with my install, worked perfectly fine

**Comment by kevinpapst** [comment]
Sorry, I was talking about:

> This can easily be fixed by setting the max-value in Entity/User.php:162 from the current 60 to the actual field length of 180

The column length is wrong due to historically reasons and I don't want to shorten it to avoid bugs.
But the limit should stay in place. 

With changing the SAML mapping code, would the limit problem be fixed as well?

**Comment by lukasassmann** [comment]
When changing the SAML mapping code one could use a different field as an identifier (we would for one prefer to user Authentik's username). However, using the default configuration would still produce the same issues.

Is there a reason the username should not be longer than 60 characters? 

I can try to implement a change to the SamlProvider and test it with our application, but i still think it should work with the currently used identifier

**Comment by kevinpapst** [comment]
> Is there a reason the username should not be longer than 60 characters?

Yes. The length of the database field comes from the very old (and not longer used) FOSUserBundle. Kimai always forced 60 chars. I don't see a reason to soften that limit without good reasoning and proper testing.
Especially not to 180 characters, if not necessary.

**Comment by kevinpapst** [comment]
How long can Authentiks username be?
Did you have a chance to look into the Mapping issue?

**Comment by lukasassmann** [comment]
The identifier is a 64 char string. I will try to provide a solution for the SAML mapping by the end of the week

**Comment by lukasassmann** [comment]
Ok, so i now had time to dig a little deeper and found that the issue was with me/my configuration. 

When using the authentik userId the 60 char max. length was an issue that should be fixed with your changes to 64 char length. 

However, the field mapping issue is not an issue at all. You can define in authentik (and pretty much all other IDPs) which of the users value should be used as the id for the authorized app. So instead of changing the mapping behaviour in kimai to a wonky "see if the config says to overwrite the username and take adjustments accordingly" kinda thing, you should just tell authentik/your IDP to use a different field. 

I just now found the setting (it's a bit hidden and I'm fairly new to authentik and its ways), changed it and it all works as expected. we now have our authentik usernames as identifiers in kimai 👍🏼

tl;dr:
- extending the max length to 64 characters was a good and necessary adjustment imo
- SAML mapping does not need to be changed
- Setup your IDP to use the correct field as identifier
- all is well :)

This issue could therefore be closed imho

**Comment by kevinpapst** [comment]
Thanks for your feedback!

Any chance you could share your findings with the community?

What should be added or improved on this page:
https://www.kimai.org/documentation/saml-authentik.html

You should be able to click on edit here:
https://github.com/kimai/www.kimai.org/blob/main/_documentation/saml-authentik.md
and then follow the workflow to save and submit a PR directly here in the Github UI.
